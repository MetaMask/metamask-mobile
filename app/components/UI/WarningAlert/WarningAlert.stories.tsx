// Third party dependencies.
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { text } from '@storybook/addon-knobs';

// Internal dependencies.
import WarningAlert from './WarningAlert';

storiesOf('Components / UI / WarningAlert', module)
  .addDecorator((getStory) => getStory())
  .add('Default', () => {
    const groupId = 'Props';
    const labelSelector = text(
      'text',
      'Text of a warning alert example',
      groupId,
    );

    return <WarningAlert text={labelSelector} dismissAlert={() => false} />;
  });
