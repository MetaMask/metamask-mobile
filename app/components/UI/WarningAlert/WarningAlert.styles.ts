// Third party dependencies.
import { StyleSheet } from 'react-native';
import { Theme } from '../../../util/theme/models';

/**
 * Style sheet function for ModalConfirmation component.
 *
 * @param params Style sheet params.
 * @param params.theme App theme from ThemeContext.
 * @returns StyleSheet object.
 */
const styleSheet = (params: { theme: Theme }) => {
  const { theme } = params;
  const { colors } = theme;

  return StyleSheet.create({
    alertContainer: {
      position: 'absolute',
      left: 16,
      right: 16,
      bottom: 20,
      borderRadius: 8,
      backgroundColor: colors.background.default,
      borderColor: colors.warning.default,
    },
    upperAlertContainer: { bottom: 120 },
    alertWrapper: {
      flex: 1,
      padding: 14,
    },

    alertText: {
      lineHeight: 18,
      color: colors.text.default,
      paddingLeft: 4,
    },
    alertIcon: {
      paddingRight: 4,
    },
  });
};

export default styleSheet;
