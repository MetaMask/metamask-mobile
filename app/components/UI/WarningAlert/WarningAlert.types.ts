export interface WarningAlertProps {
  /**
   * Warning description text
   */
  text: string;
  /**
   * useState function to unshow the warning alert
   */
  dismissAlert: () => void;
  /**
   * On press learn more action, usually to redirect to an article
   */
  onPressLearnMore?: () => void;
  /**
   * if there is a precedent alert the warning will be positioned on top
   */
  precedentAlert?: boolean;
}
