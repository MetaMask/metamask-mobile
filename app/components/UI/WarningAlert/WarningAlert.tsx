// Third party dependencies.
import React from 'react';
import { View } from 'react-native';
import MaterialCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons';

// External dependencies.
import Alert, { AlertType } from '../../Base/Alert';
import Text from '../../Base/Text';
import { useTheme } from '../../../../app/util/theme';
import { strings } from '../../../../locales/i18n';
import { useStyles } from '../../../component-library/hooks';

// Internal dependencies.
import { WarningAlertProps } from './WarningAlert.types';
import styleSheet from './WarningAlert.styles';

const WarningAlert = ({
  text,
  dismissAlert,
  onPressLearnMore,
  precedentAlert,
}: WarningAlertProps) => {
  const { colors } = useTheme();
  const { styles } = useStyles(styleSheet, {});

  return (
    <View
      style={[
        styles.alertContainer,
        precedentAlert && styles.upperAlertContainer,
      ]}
    >
      <Alert
        small
        type={AlertType.Warning}
        onDismiss={dismissAlert}
        style={styles.alertWrapper}
        renderIcon={() => (
          <MaterialCommunityIcon
            name="information"
            size={20}
            color={colors.warning.default}
            style={styles.alertIcon}
          />
        )}
      >
        <Text primary noMargin style={styles.alertText}>
          {text}
          {onPressLearnMore && (
            <Text primary noMargin link onPress={onPressLearnMore}>
              {' '}
              {strings('networks.learn_more')}
            </Text>
          )}
        </Text>
      </Alert>
    </View>
  );
};

export default WarningAlert;
