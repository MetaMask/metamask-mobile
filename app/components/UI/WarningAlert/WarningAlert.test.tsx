// Third party dependencies.
import React from 'react';
import { shallow } from 'enzyme';

// Internal dependencies.
import WarningAlert from './WarningAlert';

describe('ButtonBase', () => {
  it('should render correctly', () => {
    const mockDismissFunction = jest.fn();
    const wrapper = shallow(
      <WarningAlert text={'test'} dismissAlert={mockDismissFunction} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
