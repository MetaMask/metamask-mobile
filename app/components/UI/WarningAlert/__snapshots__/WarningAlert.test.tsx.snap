// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBase should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#F66A0A",
        "borderRadius": 8,
        "bottom": 20,
        "left": 16,
        "position": "absolute",
        "right": 16,
      },
      undefined,
    ]
  }
>
  <Alert
    onDismiss={[MockFunction]}
    renderIcon={[Function]}
    small={true}
    style={
      Object {
        "flex": 1,
        "padding": 14,
      }
    }
    type="Warning"
  >
    <Text
      noMargin={true}
      primary={true}
      style={
        Object {
          "color": "#24272A",
          "lineHeight": 18,
          "paddingLeft": 4,
        }
      }
    >
      test
    </Text>
  </Alert>
</View>
`;
